pkg/http/api.go

```diff
@@ -994,3 +1049,19 @@ func (a *api) onGetHealthz(c *routing.Context) error {
 
 	return nil
 }
+
+func getCorrelationID(c *routing.Context) string {
+
+	return string(c.Request.Header.Peek(diagnostics.CorrelationID))
+}
+
+func newContext(c *routing.Context) (context.Context, context.CancelFunc) {
+	corID := getCorrelationID(c)
+	v := TraceValues{
+		TraceID: corID,
+	}
+
+	// Set the context with the required trace values.
+	ctx := context.WithValue(c.RequestCtx, diagnostics.CtxCorrelationIDKey, &v)
```

Comment:
Can we postpone propagating correlation id here? We will need to discuss how we can handle trace id aligning to opencensus or standard way. Then We can propagate correlation id later maybe in 0.7.0 release.

Reply:
> Can we postpone propagating correlation id here? We will need to discuss how we can handle trace id aligning to opencensus or standard way. Then We can propagate correlation id later maybe in 0.7.0 release.

I think we can have this because there is no harm now and it is additional . Also the correlation id is already there in the fasthttp request context which is added through tracing layer, so it is fine.

Once we start opencensus way, then we can modify as appropriate.

Reply:
is this propagated context value used anywhere? I do not see it... We should not introduce the unused property.

Reply:
> is this propagated context value used anywhere? I do not see it... We should not introduce the unused property.

It is part of the context. It is not used because we are not logging in the trace. We need to update the log traces that we can update in different PR.